### PR TITLE
EN-1993 upgraded to spring security and removed OLTU

### PR DESCRIPTION
**Pull request by eugeniosant**
Merge only after check of the required improvement on AppBuilder project (EN-2138)

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

File: admin-console/src/main/webapp/resources/administration/js/pages/pageModel-entry.js [hunk 1]
```diff
@@ -16,10 +16,10 @@ $(function() {
 
     var match = window.location.href.match(/(^.+\/do\/)/ ),
         baseUrl = match ? match[0] : window.location.protocol + '//' + window.location.host,
-        pathnameArray = location.pathname.split(";"),
-        pathname = (pathnameArray.length > 0) ? ';'+pathnameArray[1] : '',
-        serviceUrl = baseUrl + 'rs/PageModel/frames' + pathname + '?code=' + PROPERTY.code,
-        updateUrl = baseUrl + 'rs/PageModel/updateSketch' + pathname + '?code=' + PROPERTY.code;
+        // pathnameArray = location.pathname.split(";"),
```
Comment (joewhite101) [inline admin-console/src/main/webapp/resources/administration/js/pages/pageModel-entry.js]: Remove commented out code if not needed

File: admin-console/src/main/webapp/resources/administration/js/pages/pageTreeMenu.js [hunk 2]
```diff
@@ -2,20 +2,20 @@ $(function () {
...
-            moveWidgetUrl = baseUrl + 'rs/Page/moveWidget' + pathname + '?code=' + PROPERTY.pagemodel,
-            deleteWidgetUrl = baseUrl + 'rs/Page/deleteWidget' + pathname + '?code=' + PROPERTY.pagemodel,
-            getPageDetailUrl = baseUrl + 'rs/Page/detail' + pathname + '',
-            restoreOnlineUrl = baseUrl + 'rs/Page/restoreOnlineConfig' + pathname + '',
-            setOnlineUrl = baseUrl + 'rs/Page/setOnline' + pathname + '',
-            setOfflineUrl = baseUrl + 'rs/Page/setOffline' + pathname + '',
-            configureWidgetUrl = baseUrl + 'Page/editFrame.action' + pathname + '',
-            apiMappingsUrl = baseUrl + 'rs/Portal/WidgetType/apiMappings' + pathname + '',
-            apiCopyFromWidgetUrl = baseUrl + 'Api/Service/copyFromWidget.action' + pathname + '',
+    //var pathnameArray = location.pathname.split(";");
```
Comment (joewhite101) [inline admin-console/src/main/webapp/resources/administration/js/pages/pageTreeMenu.js]: Remove commented out code if not needed

File: admin-console/src/main/webapp/resources/administration/js/pages/pageTreeMenu.js [hunk 3]
```diff
@@ -2,20 +2,20 @@ $(function () {
...
+    //var pathname = (pathnameArray.length > 0) ? ';'+pathnameArray[1] : '';
+
+    var serviceUrl = baseUrl + 'rs/PageModel/frames' + /*pathname +*/ '?code=' + PROPERTY.pagemodel,
+            addWidgetUrl = baseUrl + 'rs/Page/joinWidget' + /*pathname +*/ '?code=' + PROPERTY.pagemodel,
```
Comment (joewhite101) [inline admin-console/src/main/webapp/resources/administration/js/pages/pageTreeMenu.js]: Remove /*pathname */ if not needed

File: engine/src/main/java/com/agiletec/aps/system/services/user/AuthenticationProviderManager.java [hunk 4]
```diff
@@ -74,37 +78,33 @@ protected UserDetails extractUser(String username, String password) throws ApsSy
...
         } catch (Throwable t) {
             throw new ApsSystemException("Error detected during the authentication of the user " + username, t);
         }
         return user;
     }
 
     private void registerToken(final UserDetails user) {
-
-        OAuthIssuer oauthIssuerImpl = new OAuthIssuerImpl(new MD5Generator());
+        /*
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/user/AuthenticationProviderManager.java]: Remove

File: engine/src/main/java/org/entando/entando/aps/servlet/security/AuthorizationServerConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.security.oauth2.config.annotation.configurers.ClientDetailsServiceConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.AuthorizationServerConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer;
+import org.springframework.security.oauth2.provider.ClientDetailsService;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+
+@Configuration
+@EnableAuthorizationServer
+public class AuthorizationServerConfiguration extends AuthorizationServerConfigurerAdapter {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/servlet/security/AuthorizationServerConfiguration.java]: Make sure to add tests for this

File: engine/src/main/java/com/agiletec/aps/system/services/user/AuthenticationProviderManager.java [hunk 6]
```diff
@@ -114,58 +114,70 @@ private void registerToken(final UserDetails user) {
...
+            return;
+        }
+        for (int i = 0; i < auths.size(); i++) {
+            Authorization authorization = auths.get(i);
+            user.addAuthorization(authorization);
+        }
+    }
+
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/user/AuthenticationProviderManager.java]: Add tests

File: engine/src/main/java/org/entando/entando/aps/system/services/oauth2/ApiOAuth2TokenManager.java [hunk 7]
```diff
@@ -1,63 +1,163 @@
...
 
     private static final Logger logger = LoggerFactory.getLogger(ApiOAuth2TokenManager.class);
     private static final String ERROR_ADDING_TOKEN = "Error adding OAuth2Token";
     private transient final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
 
-    private OAuth2TokenDAO oAuth2TokenDAO;
+    private IOAuth2TokenDAO oAuth2TokenDAO;
 
-    public OAuth2TokenDAO getOAuth2TokenDAO() {
+    @Override
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/oauth2/ApiOAuth2TokenManager.java]: Test coverage for the new methods.

File: engine/src/main/java/org/entando/entando/aps/system/services/oauth2/ApiOAuth2TokenManager.java [hunk 8]
```diff
@@ -1,63 +1,163 @@
...
+        logger.warn("removeAccessTokenUsingRefreshToken Not supported yet.");
+        throw new UnsupportedOperationException("removeAccessTokenUsingRefreshToken Not supported yet.");
     }
 
     @Override
+    public OAuth2AccessToken getAccessToken(OAuth2Authentication authentication) {
+        return this.createToken(authentication, "implicit");
+    }
+
+    private OAuth2AccessToken createToken(OAuth2Authentication authentication, String grantType) {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/oauth2/ApiOAuth2TokenManager.java]: Test coverage

File: engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuth2TokenDAO.java [hunk 9]
```diff
@@ -13,36 +13,178 @@
...
+        return "api_oauth_tokens";
+    }
+
+    @Override
+    protected String getMasterTableIdFieldName() {
+        return "accesstoken";
+    }
+
+    @Override
+    public List<OAuth2AccessToken> findTokensByClientIdAndUserName(String clientId, String username) {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuth2TokenDAO.java]: Test coverage

File: engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuthConsumerManager.java [hunk 10]
```diff
@@ -16,90 +16,127 @@
...
+    @Override
     public void init() throws Exception {
-        _logger.debug("{} ready", this.getClass().getName());
+        logger.debug("{} ready", this.getClass().getName());
     }
 
-    protected void release() {
-        super.release();
+    @Override
+    public ClientDetails loadClientByClientId(String clientId) throws ClientRegistrationException {
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuthConsumerManager.java]: Test coverage

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: admin-console/src/main/webapp/resources/administration/js/pages/pageModel-entry.js (hunk 1 above)
Comment (eugeniosant) [reply admin-console/src/main/webapp/resources/administration/js/pages/pageModel-entry.js]: Done

File: admin-console/src/main/webapp/resources/administration/js/pages/pageTreeMenu.js (hunk 2 above)
Comment (eugeniosant) [reply admin-console/src/main/webapp/resources/administration/js/pages/pageTreeMenu.js]: Done

File: admin-console/src/main/webapp/resources/administration/js/pages/pageTreeMenu.js (hunk 3 above)
Comment (eugeniosant) [reply admin-console/src/main/webapp/resources/administration/js/pages/pageTreeMenu.js]: Done

File: engine/src/main/java/com/agiletec/aps/system/services/user/AuthenticationProviderManager.java (hunk 4 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/user/AuthenticationProviderManager.java]: Done

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by pietrangelo** [comment]
retest this please

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/com/agiletec/aps/system/services/user/AuthenticationProviderManager.java (hunk 6 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/user/AuthenticationProviderManager.java]: Done

File: engine/src/main/java/org/entando/entando/aps/servlet/security/AuthorizationServerConfiguration.java (hunk 5 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/servlet/security/AuthorizationServerConfiguration.java]: Done

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuthConsumerManager.java (hunk 10 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuthConsumerManager.java]: Done

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/services/oauth2/ApiOAuth2TokenManager.java (hunk 7 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/oauth2/ApiOAuth2TokenManager.java]: Added tests for TokenDao and TokenManager

File: engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuth2TokenDAO.java (hunk 9 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/oauth2/OAuth2TokenDAO.java]: Done

File: engine/src/main/java/org/entando/entando/aps/system/services/oauth2/ApiOAuth2TokenManager.java (hunk 8 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/oauth2/ApiOAuth2TokenManager.java]: Done

**Pull request merged**